addons/website_forum/models/forum.py

```diff
@@ -364,16 +394,12 @@ def _is_self_reply(self):
         for post in self:
             post.self_reply = post.parent_id.create_uid.id == post._uid
 
-    @api.depends('child_ids.create_uid', 'website_message_ids')
+    @api.depends('child_ids.create_uid')
     def _get_child_count(self):
-        def process(node):
-            total = len(node.website_message_ids) + len(node.child_ids)
-            for child in node.child_ids:
-                total += process(child)
-            return total
-
+        requests_data = self.env['forum.post'].read_group([('parent_id', 'in', self._ids)], ['parent_id'], ['parent_id'])
```

Comment:
I didn't compare, I didn't know I should've done that, and don't know how to do it.
I followed a previous review btw.

I don't understand the "add index on parent_id"

Reply:
> add index on parent_id:


Since we are adding a lot of request based on the parent_id, it make sense to add an index on this field in postgresql. Adding index=True on the field declaration.
`parent_id  = fields....(xxx, index=True)`


>  "I didn't know I should've done that, and don't know how to do it.
> I followed a previous review btw.

Open your mind ;) you need to measure the cost between a read_group to find 1 post, and the cost of search for 1 post.  Probably that 1 search is faster than one read_group, but if each time we use forum.post, we do  20 post at the same time, are you sure that 20 x [time of 1 search] is still < than 1x [time of readgroup].
I don't have the reply, it is why I ask you your opinion and try to be sure that you make the job to think about it.

It is also one thing to add in your competence. In RD you need to have a part of research and analyze... to code something that make sense ;)

Reply:
ps: maybe @Feyensv already check for this one ? I See that you touch on this one too in a recent commit

Reply:
If we drop `website_message_ids` from the count total, the compute is highly simplified, and its performance can be improved, but I've the impression that you're missing something if you use a read_group here.

If I'm not wrong, by using a `read_group` (grouping by `parent_id`) you're losing the recursive part of the count, only considering the direct `parent_id`.   Note that I'm not sure how the behavior works functionally (maybe it cannot happen), but from the code, I suppose it can happen.

Expl:
Post A: 2 children B & C
Post B: 1 child D
Post C: 1 child E

Before your changes:
`A.child_count = 4
`
After your changes:
`A.child_count = 2
`

If it is intended, the read_group looks good to me.  Otherwise, you probably have to keep a search_count for each record (Like I did in https://github.com/odoo/odoo/pull/44662).

```python
        for post in self:
            post.child_count = self.search_count([('id', 'child_of', post.id)])
```

Reply:
if i'm not wrong, there are no recusrivity in response.

You have a question (parent_id = False), and all the reply (parent_id = question).
You cannot reply to a reply.
And this task remove the previous complexity since the comments (old mail.massage) are no more considered in the count of answer.

So we are back to initial question...

1 read_group (as well for 1 post as well for 20 post) or one search_count by post ? 
Did you try to make some benchmark with a real database ?

My first idea will be a read_group. But from old test, sometime it is better 200 search_count than 1 read_group... So on small table, we keep search_count (forum.orum e.g.). But in case of a 'biggest' table like that, I don't have real opinion, so maybe you already do the job to compare it?

Reply:
> if i'm not wrong, there are no recusrivity in answer.

Ok, then I've nothing against the `read_group`.

> 1 read_group (as well for 1 post as well for 20 post) or one search_count by post ?
Did you try to make some benchmark with a real database ?

My task was mainly to remove the len(rec.x2mfield) from count computes, replacing them by a read_group for queries reduction and computation perfs (FP request :p).  But I didn't consider that 200 search_count could sometimes be better than 1 read_group :thinking:.
I'll investigate it next week (I'm on "holiday" currently) to better understand which one is better, depending on the models, and maybe adapt my recent commits to potentially replace some `read_group`'s by `search_count`'s.

If this task has to go forward before that and you do some testing, I'd like to hear the results of it.

Thanks for the ping and advice :+1:.